### ISSUE TITLE
Relax constraint on main branch required to pass validation script

### ISSUE THREAD

**Issue by obriensystems** [terraform-google-modules/terraform-example-foundation#964]
### TL;DR
for
https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/0-bootstrap/README.md#deploying-with-cloud-build

Just curious as to why we are requiring "main" as the branch to pass validation
I could understand a requirement on "master" so we dont have to 'git checkout dev" during automation but there is no main branch.
For local development are we required to to use "main"
minor - just trying to understand the thinking behind this

https://github.com/terraform-google-modules/terraform-example-foundation/commit/84bbd257095152babdc143a6381a48fd70dbe375#diff-87c85f89e05563a983f50915c26ee6c569d2a74e13bff0d26dc9f7d740968a11R124

for 
https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/scripts/validate-requirements.sh#L126

part of day0 install ops in #940 

### Expected behavior

pass validation for any dev branch

### Observed behavior

main branch required

### Terraform Configuration

```hcl
root_@cloudshell:~/lz-tef-plz/CloudLandingZone (lz-tef-plz)$ terraform version
Terraform v1.4.6
on linux_amd64
root_@cloudshell:~/lz-tef-plz/CloudLandingZone (lz-tef-plz)$ gcloud version
Google Cloud SDK 428.0.0
root_@cloudshell:~/lz-tef-plz/CloudLandingZone (lz-tef-plz)$ git version
git version 2.30.2

all good (above 1.3.0, 393, 2.28)
```

### Terraform Version

```sh
root_@cloudshell:~/lz-tef-plz/CloudLandingZone (lz-tef-plz)$ terraform version
Terraform v1.4.6
on linux_amd64
root_@cloudshell:~/lz-tef-plz/CloudLandingZone (lz-tef-plz)$ gcloud version
Google Cloud SDK 428.0.0
root_@cloudshell:~/lz-tef-plz/CloudLandingZone (lz-tef-plz)$ git version
git version 2.30.2

all good (above 1.3.0, 393, 2.28)
```

### Additional information

all past the defaults
```
root_@cloudshell:~/lz-tef-plz/CloudLandingZone/terraform-example-foundation/0-bootstrap (lz-tef-plz)$ export EMAIL=$(gcloud config list --format json|jq .core.account | sed 's/"//g')

../scripts/validate-requirements.sh -o $ORGANIZATION_ID -b $BILLING_ID -u $EMAIL

root_@cloudshell:~/lz-tef-plz/CloudLandingZone/terraform-example-foundation/0-bootstrap (lz-tef-plz)$ ../scripts/validate-requirements.sh -o $ORGANIZATION_ID -b $BILLING_ID -u $EMAIL
Validating required utility tools...
Validating Terraform installation...
Validating Google Cloud SDK installation...
Validating Git installation...
  git default branch must be configured as main.
  See the instructions at https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/docs/TROUBLESHOOTING.md#default-branch-setting .
Validating local gcloud configuration...
Validating roles assignment for current end user credential...
Validating 0-bootstrap configuration...
.......................................
Validation failed!
Errors found:
  git default branch must be configured as main.
```

**Comment by obriensystems** [comment]
line 129 
        #ERRORS+=$'  git default branch must be configured as main.\n'
```
root_@cloudshell:~/lz-tef-plz/CloudLandingZone/terraform-example-foundation/0-bootstrap (lz-tef-plz)$ ../scripts/validate-requirements.sh -o $ORGANIZATION_ID -b $BILLING_ID -u $EMAIL
Validating required utility tools...
Validating Terraform installation...
Validating Google Cloud SDK installation...
Validating Git installation...
  git default branch must be configured as main.
  See the instructions at https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/docs/TROUBLESHOOTING.md#default-branch-setting .
Validating local gcloud configuration...
Validating roles assignment for current end user credential...
Validating 0-bootstrap configuration...
.......................................
Validation successful!
No errors found.
```

**Comment by obriensystems** [comment]
Also document in
https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/docs/TROUBLESHOOTING.md#default-branch-setting

 under which circumstances we must be in main - i suspect this is later for one or more of the CSR repos - not the repo cloned from github (only master there)

such as
https://source.cloud.google.com/prj-b-cicd-anjl/gcp-policies/+/main:

**Comment by fmichaelobrien** [comment]
Assigned to fmichaelobrien
